### ISSUE TITLE
aws_budgets_budget cannot rename budget name

### ISSUE THREAD

**Issue by heycasey** [terraform-providers/terraform-provider-aws#4616]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v0.11.7
+ provider.aws v1.19.0
```

### Affected Resource(s)

* aws_budgets_budget

### Terraform Configuration Files

```hcl
resource "aws_budgets_budget" "account" {
  name = "monthly_5k"
  budget_type = "COST"
  limit_amount = "5000"
  limit_unit = "USD"
  time_period_start = "2018-01-01_00:00"
  time_unit = "MONTHLY"
}
```

### Crash Behavior

```
Error: Error applying plan:

1 error(s) occurred:

* aws_budgets_budget.account: 1 error(s) occurred:

* aws_budgets_budget.account: update budget failed: NotFoundException: [Exception=NotFoundException]Failed to call UpdateBudget - Unable to get budget: monthly_5k_2 - the budget doesn't exist.
        status code: 400, request id: 9875d75d-5e0f-11e8-ad7a-899141acb82c
```

### Expected Behavior

The budget should be updated to reflect the new name.

### Actual Behavior

Crash with the output displayed above.

### Steps to Reproduce

- Run `terraform apply` to create the budget.
- Rename the `name` argument to anything else. In this case, I renamed it to `monthly_5k_2`
- Re-run `terraform apply`

### Important Factoids

Given the nature of the `CreateBudget` and `UpdateBudget` API behavior, perhaps a budget rename requires the existing resource to be deleted and a new one created?

**Comment by saravanan30erd** [comment]
@heycasey Correct, `name` can't be edited and it should be deleted and recreated.

**Comment by saravanan30erd** [comment]
PR submitted #4656

**Comment by bflad** [comment]
Thanks for this report @heycasey and for the fix @saravanan30erd -- #4656 has been merged and will release with v1.21.0 of the AWS provider, likely middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.21.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1210-may-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
